From 2b374b4802841a53a9e0d13ce18c38f3556bdee3 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Thu, 6 Feb 2025 00:43:05 +0100
Subject: [PATCH 1603/1695] drivers: stm32_exti: drop "wakeup-parent" magic
 value

With each use of "wakeup-parent" magic value already dropped, drop
also the related code to temporarily handle it.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 56935adca7ca545b056f891d0d8f742e4ae1f092 drivers: stm32_exti: drop "wakeup-parent" magic value]
Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I510d1c1f878574fcf5ca2b9b1bcbf5343b9d107e
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/434945
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/stm32_exti.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/core/drivers/stm32_exti.c b/core/drivers/stm32_exti.c
index 63e3bf566..51f21bc85 100644
--- a/core/drivers/stm32_exti.c
+++ b/core/drivers/stm32_exti.c
@@ -719,7 +719,6 @@ TEE_Result stm32_exti_get_pdata(const void *fdt, int nodeoffset,
 }
 
 /* Callback for "interrupts" and "interrupts-extended" DT node properties */
-/* FIXME: currently handling "wakeup-parent" too */
 static TEE_Result
 stm32_exti_dt_get_chip_cb(struct dt_pargs *pargs, void *priv_data,
 			  struct itr_desc *itr_desc)
@@ -739,10 +738,6 @@ stm32_exti_dt_get_chip_cb(struct dt_pargs *pargs, void *priv_data,
 	itr_desc->chip = &exti->chip;
 	itr_desc->itr_num = exti_line;
 
-	/* FIXME: "wakeup-parent" magic value. To be removed */
-	if (exti_line == 0xffff)
-		return TEE_SUCCESS;
-
 	if (exti_line >= stm32_exti_nbevents(exti))
 		return TEE_ERROR_GENERIC;
 
-- 
2.47.2

